autorest/azure/async.go

```
@@ -422,6 +427,7 @@ func (pt *pollingTrackerBase) pollForStatus(sender autorest.Sender) error {
 
 // attempts to unmarshal a ServiceError type from the response body.
 // if that fails then make a best attempt at creating something meaningful.
+// NOTE: this assumes that the async operation has failed.
 func (pt *pollingTrackerBase) updateErrorFromResponse() {
```

>updateErrorFromResponse [](start = 30, length = 23)

can we return an error if it doesnt make sense to call this ?